python/lsst/scarlet/lite/image.py

```diff
@@ -0,0 +1,1372 @@
+                if len(indices) > 1 and indices[1] in self.bands:
+                    # The indices are all band names,
+                    # so use them all as a spectral indices
+                    bands = indices
+                    spectral_index = self.spectral_indices(bands)
+                    y_index = x_index = slice(None)
+                elif self._is_spectral_index(indices[0]):
+                    # The first index is a spectral index
+                    spectral_index = self.spectral_indices(indices[0])
+                    if isinstance(spectral_index, slice):
+                        bands = self.bands[spectral_index]
+                    elif len(spectral_index) == 1:
+                        bands = ()
+                        spectral_index = spectral_index[0]  # type: ignore
+                    else:
+                        bands = tuple(self.bands[idx] for idx in spectral_index)
+                    indices = indices[1:]
+                    if len(indices) == 1:
+                        # The spatial index must be a bounding box
+                        if not isinstance(indices[0], Box):
+                            raise IndexError(f"Expected a Box for the spatial index but got {indices[1]}")
+                        y_index, x_index = self._get_box_slices(indices[0])
+                    elif len(indices) == 0:
+                        y_index = x_index = slice(None)
+                    else:
+                        raise IndexError(f"Too many spatial indices, expeected a Box bot got {indices}")
+                full_index = (spectral_index, y_index, x_index)
+            elif isinstance(indices[0], Box):
+                bands = self.bands
+                y_index, x_index = self._get_box_slices(indices[0])
+                full_index = (slice(None), y_index, x_index)
+            else:
+                error = f"3D images can only be indexed by spectral indices or bounding boxes, got {indices}"
+                raise IndexError(error)
+        else:
+            if len(indices) != 1 or not isinstance(indices[0], Box):
+                raise IndexError(f"2D images can only be sliced by bounding box, got {indices}")
+            bands = ()
+            y_index, x_index = self._get_box_slices(indices[0])
+            full_index = (y_index, x_index)  # type: ignore
+
+        y0 = y_index.start
+        if y0 is None:
+            y0 = 0
+
+        x0 = x_index.start
+        if x0 is None:
+            x0 = 0
+
+        if value is None:
+            # This is a getter,
+            # so return an image with the data sliced properly
+            yx0 = (y0 + self.yx0[0], x0 + self.yx0[1])
+
+            data = self.data[full_index]
+
+            if len(data.shape) == 2:
+                # Only a single band was selected, so return that band
+                return Image(data, yx0=yx0)
+            return Image(data, bands=bands, yx0=yx0)
+
+        # Set the data
+        self._data[full_index] = value.data
```

Comment:
Why are you setting values in a getter?

Reply:
This isn't the getter. I didn't want to have to duplicate code, especially if the API changed, so I implemented a single private method that both the getter and setter call. But if you'd like me to rename it I can.